gems/aws-sdk-core/lib/aws-sdk-sts/presigner.rb

```diff
@@ -0,0 +1,73 @@
+require 'aws-sigv4'
+
+module Aws
+  module STS
+    # Allows you to create presigned URLs for STS operations.
+    #
+    # Example Use:
+    #
+    #   signer = Aws::STS::Presigner.new
+    #   url = signer.get_caller_identity_presigned_url(
+    #     headers: {"X-K8s-Aws-Id" => 'my-eks-cluster'}
+    #   )
+    #
+    class Presigner
+      # @option options [Client] :client Optionally provide an existing
+      #   STS client
+      def initialize(options = {})
+        @client = options[:client] || Aws::STS::Client.new
+      end
+
+      # Returns a presigned url for get_caller_identity.
+      #
+      # @option options [Hash] :headers
+      #   Headers that should be signed and sent along with the request. All
+      #   x-amz-* headers must be present during signing. Other headers are
+      #   optional.
+      #
+      # @option options [Integer] :expires_in (60) Number of seconds before
```

Comment:
Does x-amz-expires actually work for STS?,  the documentation link in Presigner refers to S3 and why 60 as a default?

Reply:
It appears you are correct. I have overlooked that in the code review. Sorry.. 

Docs: https://docs.aws.amazon.com/STS/latest/APIReference/CommonParameters.html

I can remove this from the PR and update the documentation.

Reply:
I found this issue: https://github.com/aws/aws-sdk-go/issues/2167 that suggests STS has a 15 minute expiration and will not be changing (last year).